Sources/XcodeGenKit/SchemeGenerator.swift

```diff
@@ -241,12 +241,19 @@ public class SchemeGenerator {
             storeKitConfigurationFileReference = XCScheme.StoreKitConfigurationFileReference(identifier: storeKitConfigurationPath.string)
         }
 
+        let macroExpansion: XCScheme.BuildableReference?
+        if let macroExpansionName = scheme.run?.macroExpansion {
+            macroExpansion = buildActionEntries.first(where: { $0.buildableReference.blueprintName == macroExpansionName })?.buildableReference ?? buildableReference
+        } else {
+            macroExpansion = shouldExecuteOnLaunch ? nil : buildableReference
```

Comment:
I can't remember the exact reason for this existing logic, but should it be applied if we have a `macroExpansion`?

Reply:
I also have no idea but it would be better to fix 246 like below to keep back compatibility of it. 

```swift
if let macroExpansionName = scheme.run?.macroExpansion,
   let resolvedMacroExpansion = shouldExecuteOnLaunch ? nil : buildActionEntries.first(where: { $0.buildableReference.blueprintName == macroExpansionName })?.buildableReference {
    macroExpansion = resolvedMacroExpansion
} else {
    // change to obey as is implementation
    macroExpansion = shouldExecuteOnLaunch ? nil : buildableReference
}
```

What do you think?

Reply:
I pushed new code same with my comment, please check it.